src/art/model.py

```diff
@@ -52,6 +100,38 @@ async def register(self, api: "LocalAPI") -> None:
         self._api = api
         await self._api.register(self)
 
+    def openai_client(
+        self,
+    ) -> AsyncOpenAI:
+        if self.inference_api_key is None or self.inference_base_url is None:
+            raise ValueError(
+                "OpenAI client not yet available. You must call `model.register()` first."
+            )
+        openai_client = AsyncOpenAI(
```

Comment:
Rather than recreating this client for each request, I'd rather use a singleton pattern (define the client once, check if its defined, reuse if present) to reduce CPU overhead and time spent on TLS handshakes.

(see my [reference implementation](https://github.com/OpenPipe/ART/pull/73/files#r2060619101))

Reply:
Fair, updated to store. It's unclear to me whether there's any extra TLS handshakes from creating the client every time but possible there are.